code/modules/mob/living/carbon/human/ai/brain/ai_brain_helpers.dm

```diff
@@ -0,0 +1,128 @@
+	current_order.brains += src
+
+/datum/human_ai_brain/proc/remove_current_order()
+	if(current_order)
+		current_order.brains -= src
+	current_order = null
+
+/// Returns TRUE if the target is friendly/neutral to us
+/datum/human_ai_brain/proc/faction_check(atom/target)
+	if(isdefenses(target))
+		var/obj/structure/machinery/defenses/defense_target = target
+		if(tied_human.faction in defense_target.faction_group)
+			return TRUE
+		return FALSE
+
+	if(ismob(target))
+		var/mob/mob_target = target
+
+		if(mob_target.faction == tied_human.faction)
+			return TRUE
+
+		if(mob_target.faction in friendly_factions)
+			return TRUE
+
+		if(mob_target.faction in neutral_factions)
+			return TRUE
+
+	return FALSE
+
+/datum/human_ai_brain/proc/on_neutral_faction_betray(faction)
+	if(!tied_human.faction)
+		return
+
+	var/datum/human_ai_faction/our_faction = SShuman_ai.human_ai_factions[tied_human.faction]
+	if(!our_faction)
+		return
+
+	our_faction.remove_neutral_faction(faction)
+	our_faction.reapply_faction_data()
+
+/datum/human_ai_brain/proc/on_handcuffed(datum/source)
+	SIGNAL_HANDLER
+
+	if((tied_human.stat >= DEAD) || tied_human.client)
+		return
+
+	message_admins("AI human [tied_human.real_name] has been handcuffed while alive or unconscious.", tied_human.x, tied_human.y, tied_human.z)
+
+/datum/human_ai_brain/proc/ensure_primary_hand(obj/item/held_item)
+	if(tied_human.get_inactive_hand() == held_item)
+		tied_human.swap_hand()
+
+/datum/human_ai_brain/proc/unholster_primary()
+	if(!primary_weapon || tied_human.l_hand == primary_weapon || tied_human.r_hand == primary_weapon)
+		return
+
+	var/cur_hand = tied_human.get_active_hand()
+	if(cur_hand)
+		tied_human.drop_held_item(cur_hand)
+
+	tied_human.u_equip(primary_weapon)
+	tied_human.put_in_active_hand(primary_weapon)
+
+	primary_weapon.guaranteed_delay_time = world.time
+	primary_weapon.wield_time = world.time
+	primary_weapon.pull_time = world.time
+
+/datum/human_ai_brain/proc/wield_primary()
+	primary_weapon?.wield(tied_human)
+
+/datum/human_ai_brain/proc/wield_primary_sleep()
+	wield_primary()
+	sleep(max(primary_weapon?.wield_delay, short_action_delay * action_delay_mult))
+
+/datum/human_ai_brain/proc/holster_primary()
+	if(tied_human.s_store || (tied_human.l_hand != primary_weapon && tied_human.r_hand != primary_weapon))
+		return FALSE
+
+	return tied_human.equip_to_slot_if_possible(primary_weapon, WEAR_J_STORE, TRUE)
+
+/datum/human_ai_brain/proc/unholster_melee()
+	if(istype(tied_human.l_hand, /obj/item) || istype(tied_human.r_hand, /obj/item))
+		return TRUE
+
+	var/cur_hand = tied_human.get_active_hand()
+	if(cur_hand)
+		tied_human.drop_held_item(cur_hand)
+
+	if(tied_human.shoes)
+		return tied_human.shoes.remove_item(tied_human)
+
+/datum/human_ai_brain/proc/unholster_any_weapon()
+	if(unholster_melee())
+		return
+	if(primary_weapon)
+		unholster_primary()
+		ensure_primary_hand(primary_weapon)
+		return
+	// insert any viable weapon slot macros in here
```

Comment:
wdym by this

Reply:
with unholster_any_weapon() we try and get ANY available weapon in our hands, which is used when we try to shatter a glass or destroy a resin structure in the way. Preferrably we should insert any procs that unholsters stuff in here, so AI will try and use any possibility to break things with something rather than it's bare hands